openmmtools/alchemy.py

```diff
@@ -1582,43 +1691,59 @@ def _alchemically_modify_HarmonicBondForce(reference_force, alchemical_region):
-        custom_force.addGlobalParameter('lambda_bonds', 1.0)
+        custom_force.addGlobalParameter('lambda_bonds_{}'.format(alchemical_region.name), 1.0)
         custom_force.addPerBondParameter('r0')
         custom_force.addPerBondParameter('K')
         # Process reference bonds.
-        for bond_index in range(reference_force.getNumBonds()):
+        for bond_index in range(force.getNumBonds()):
             # Retrieve parameters.
-            [particle1, particle2, theta0, K] = reference_force.getBondParameters(bond_index)
+            [particle1, particle2, theta0, K] = force.getBondParameters(bond_index)
             if bond_index in alchemical_region.alchemical_bonds:
-                # Alchemically modified bond.
+                # add alchemical
                 custom_force.addBond(particle1, particle2, [theta0, K])
-            else:
-                # Standard bond.
-                force.addBond(particle1, particle2, theta0, K)
+                #remove from std
+                force.setBondParameters(bond_index, particle1, particle2, theta0, K*0.0)
+
+        return {'': [force], 'lambda_bonds_{}'.format(alchemical_region.name): [custom_force]}
+
+    def _get_nonbonded_energy_exspressions(self, alchemical_region_idx):
+        # Sterics mixing rules.
+        sterics_mixing_rules = ('epsilon = sqrt(epsilon1*epsilon2);'  # Mixing rule for epsilon.
+                                'sigma = 0.5*(sigma1 + sigma2);')  # Mixing rule for sigma.
+
+        # Soft-core Lennard-Jones.
+        exceptions_sterics_energy_expression = ('U_{0}_sterics;'
+                                                'U_{0}_sterics = (lambda_sterics_{0}^softcore_a)*4*epsilon*x*(x-1.0);'
+                                                'x = (sigma/reff_sterics)^6;'
+                                                # Effective softcore distance for sterics.
+                                                'reff_sterics = sigma*((softcore_alpha*(1.0-lambda_sterics_{0})^softcore_b + (r/sigma)^softcore_c))^(1/softcore_c);').format(
+                                                alchemical_region_idx)
 
-        return {'': [force], 'lambda_bonds': [custom_force]}
 
-    def _get_electrostatics_energy_expressions(self, reference_force):
+        # Define energy expression for electrostatics.
+        return sterics_mixing_rules, exceptions_sterics_energy_expression
+
+    def _get_electrostatics_energy_expressions(self, reference_force, alchemical_region_idx):
         """Return the energy expressions for electrostatics."""
 
         # The final expression will be prefix + method + suffix.
-        electrostatics_prefix = ('U_electrostatics;'
-                                 'U_electrostatics=(lambda_electrostatics^softcore_d)*ONE_4PI_EPS0*chargeprod')
+        electrostatics_prefix = ('U_{0}_electrostatics;'
+                                 'U_{0}_electrostatics=(lambda_electrostatics_{0}^softcore_d)*ONE_4PI_EPS0*chargeprod').format(alchemical_region_idx)
```

Comment:
I believe you can avoid adding the name of the region to `U_electrostatics`. It'll be a local variable in the kernel anyway so it's not a problem if there are duplicate variables in different forces.